daemon/config.go

```diff
@@ -11,42 +21,69 @@ const (
 	disableNetworkBridge = "none"
 )
 
+// LogConfig represents the default log configuration.
+// It includes json tags to deserialize configuration from a file
+// using the same names that the flags in the command line uses.
+type LogConfig struct {
+	Type   string            `json:"log-driver,omitempty"`
+	Config map[string]string `json:"log-opt,omitempty"`
+}
+
+// CommonTLSOptions defines TLS configuration for the daemon server.
+// It includes json tags to deserialize configuration from a file
+// using the same names that the flags in the command line uses.
+type CommonTLSOptions struct {
+	CAFile   string `json:"tlscacert,omitempty"`
+	CertFile string `json:"tlscert,omitempty"`
+	KeyFile  string `json:"tlskey,omitempty"`
+}
+
 // CommonConfig defines the configuration of a docker daemon which are
 // common across platforms.
+// It includes json tags to deserialize configuration from a file
+// using the same names that the flags in the command line uses.
 type CommonConfig struct {
-	AuthZPlugins  []string // AuthZPlugins holds list of authorization plugins
-	AutoRestart   bool
-	Bridge        bridgeConfig // Bridge holds bridge network specific configuration.
-	Context       map[string][]string
-	DisableBridge bool
-	DNS           []string
-	DNSOptions    []string
-	DNSSearch     []string
-	ExecOptions   []string
-	ExecRoot      string
-	GraphDriver   string
-	GraphOptions  []string
-	Labels        []string
-	LogConfig     container.LogConfig
-	Mtu           int
-	Pidfile       string
-	RemappedRoot  string
-	Root          string
-	TrustKeyPath  string
+	AuthZPlugins  []string            `json:"authz-plugin,omitempty"` // AuthZPlugins holds list of authorization plugins
```

Comment:
before I forget: be prepared to change this to `authorization-plugin` if #19279 gets merged